File: Algorithms/Strings/HackerRank in a String!.cs
using System;

class Solution
{
    static string hackerrankInString(string s) {
        string hackerrank = "hackerrank";
        int index = 0;
        for (int i = 0; i < s.Length; i++)
        {
            if (s[i] == hackerrank[index])
                index++;
            if (index == 9) return "YES";
        }
        return "NO";

    }
    public static void Main(string[] args)
    {
        
        int t = Convert.ToInt32(Console.ReadLine());
        while(t-- > 0)
        {
            string s = Console.ReadLine();
            Console.WriteLine(hackerrankInString(s));     
        }
    }
}

File: Algorithms/Strings/HackerRank in a String!.py
hackerrank = "hackerrank"
t = int(input())
for _ in range(t):
    string = input()
    i = 0
    for ch in string :
        if ch == hackerrank[i]:
            i+=1
            if i == 9:
                break
    if i == 9:
        print("YES")
    else:
        print("NO")
